### PR TITLE
Add report verification explainer

### PR DESCRIPTION
**Pull request by alexmturner**

### CODE REVIEW

**Comment by csharrison** [comment]
I can review sure.

File: report_verification.md [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+allow the reporting origin to partition its reports into 7 buckets.
+
+##### Potential mitigations
+
+We could consider mitigations in the future. For example:
+
+- restricting the public/private metadata to one bucket – or just a single
+  private bit to avoid an invalid traffic oracle.
+- refusing to send reports to reporting origins using report verification if no
+  token was available/issued.
```
Comment (csharrison) [inline report_verification.md:414]:
```suggestion
  token was available/issued.
- sending null reports with some frequency for buyers that delegate to an issuer who issued tokens.
```

This is pretty speculative but I think we could think of something moderately workable.

File: report_verification.md [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+
+##### Difficult to decide on number of tokens to issue
+
+Due to this large time delay between token issuance and last possible use, it
+will be difficult to decide on the number of tokens to issue. If too few are
+issued, later auctions may not be able to be attested. Issuing too many may
+degrade performance, e.g. unnecessarily using storage, and may exacerbate token
+exfiltration attacks.
+
+##### Requires new token storage
```
Comment (csharrison) [inline report_verification.md:462]:
```suggestion
##### Requires new persistent token storage
```

File: report_verification.md [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+
+### Details
+
+As it uses the same token source, most details match the FLEDGE bidder
+discussion (see [above](#details-2)). Additional considerations are listed
+below.
+
+#### Doesn’t support nesting
+
+This proposal does not support cross-origin subframes or nested fenced frames
```
Comment (csharrison) [inline report_verification.md:540]:
```suggestion
This proposal currently does not support cross-origin subframes or nested fenced frames
```

File: report_verification.md [hunk 4]
```diff
@@ -0,0 +1,556 @@
...
+something about when the report was triggered. This is partially mitigated by
+the existing randomized reporting delay (10-60 min) imposed as FLEDGE auctions
+impose small timeouts (e.g. 0.5 s). As discussed
+[above](#privacy-considerations) for Shared Storage, we avoid concerns about
+partitioning by making the number of reports deterministic (and other
+protections).
+
+## FLEDGE bidders
+
+We can’t easily use a contextual ID for the FLEDGE bidder case as the existence
```
Comment (csharrison) [inline report_verification.md:222]:
```suggestion
**Note: Unlike the above sections which offer relatively straightforward approaches,
this section is highly complex and nuanced. Feedback is appreciated!**

We can’t easily use a contextual ID for the FLEDGE bidder case as the existence
```

**Review by csharrison — APPROVED** [review]
Left a few minor comments but overall LGTM

File: report_verification.md [hunk 5]
```diff
@@ -0,0 +1,556 @@
...
+
+```jsonc
+"context_id" : "1234567890",
+```
+
+This would be achieved by adding a new optional parameter to the Shared Storage
+`run()` and `selectURL()` APIs, e.g.:
+
+```js
+sharedStorage.run("someOperation", {"contextId": 1234567890n});
```
[the remark this answers is not in the document]
Comment (alexmturner) [reply report_verification.md]: Yep, was considering doing it in a follow up, but may as well change it now. Went with "example_string" just to emphasize that it doesn't need to be a number. (Also switched js strings to single quotes to align.)

File: report_verification.md [hunk 6]
```diff
@@ -0,0 +1,556 @@
...
+### Existing threat practicality
+
+As with the Attribution Reporting API, we don’t currently prevent reports being
+created out of thin air, but practical attacks are challenging. More details
+[here](https://github.com/WICG/attribution-reporting-api/blob/main/report_verification.md#existing-mitigations-and-practical-threats).
+
+## Shared Storage
+
+When triggering a Shared Storage operation that could send an aggregatable
+report, we propose allowing the site to specify a high-entropy (64-bit) ID from
```
[the remark this answers is not in the document]
Comment (alexmturner) [reply report_verification.md]: Should we specify the length cap or leave it open in the explainer?

File: report_verification.md [hunk 7]
```diff
@@ -0,0 +1,556 @@
...
+
+We can’t easily use a contextual ID for the FLEDGE bidder case as the existence
+of a bidder in a particular auction is inherently cross-site data, see
+[below](#specifying-a-contextual-id-and-each-possible-ig-owner). So, our options
+are more limited and we focus on mechanisms using Private State Tokens.
+
+However, note also that there are no network requests that we can easily reuse
+for token issuance. While there is a trusted signals fetch, that is
+intentionally uncredentialed. Much like using an ID, we can’t just add a network
+request for each bidder as that would reveal cross-site data.
```
[the remark this answers is not in the document]
Comment (alexmturner) [reply report_verification.md:233]: I think this is true even if `interestGroupBuyers` isn't wildcarded. (Also apparently the wildcard option is not actually implemented.)

File: report_verification.md (hunk 1 above)
Comment (alexmturner) [reply report_verification.md:418]: Yeah, sg to add

**Comment by alexmturner** [comment]
Thanks, addressed/replied to comments

File: report_verification.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (csharrison) [reply report_verification.md]: I think it's too in the weeds

File: report_verification.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (csharrison) [reply report_verification.md:233]: Why? `interestGroupBuyers` is specified at auction time, so it's fine to send a request to each of them. The only concern is if we don't know the set of all buyers (which occurs when it is `*`).

File: report_verification.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (alexmturner) [reply report_verification.md]: ack

File: report_verification.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (alexmturner) [reply report_verification.md:233]: (see "Specifying a contextual ID and each possible IG owner" below)

**Comment by csharrison** [comment]
sLGTM

**Comment by alexmturner** [comment]
Thanks!

File: report_verification.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (jdfreder) [reply report_verification.md]: When you decide on a length limit, where will that be published? Inquiring minds want to know... :)

[the remark this answers is not in the document]
Comment (csharrison) [reply report_verification.md]: It should be in the spec.

[the remark this answers is not in the document]
Comment (jdfreder) [reply report_verification.md]: Thanks! I'll subscribe to the repo so that I do not miss that

**Pull request merged**